### PR TITLE
Cmpsub

### PR DESCRIPTION
**Pull request by DarrenStahl**
Added Cmp and Sub, fixed some glitches in existing code including modrm checking, and subtract overflow flag setting.

### CODE REVIEW

File: src/Instruction.cpp [hunk 1]
```diff
@@ -79,10 +79,26 @@ bool Instruction::Parity(unsigned int parity) {
 bool Instruction::OverflowSub(unsigned int val, unsigned int dst, unsigned int src, unsigned int size) {
 
 	unsigned int msb = 1 << (8 * size - 1);
-	if(dst & msb && ~(src & msb) && ~(val & msb)) {
+
+	if ((src & msb != 0) ? ~src + 1 : src > (dst & msb != 0) ? dst + 1 : dst) {
```
Comment (rdnelson) [inline src/Instruction.cpp:83]: What does this fix?

Comment (DarrenStahl) [inline src/Instruction.cpp:83]: This is checking which one is larger (absolute value) to check the overflow flag. It wasn't setting correctly in edge cases before this. (two negative numbers causing sign wrap and not setting flag). This has worked for all my test cases.

Comment (rdnelson) [inline src/Instruction.cpp:83]: Ok, I'll let it in, but I've found a much cleaner and more easy to understand version that I'll replace it with. This may work, but it's confusing as hell.

I'll also add in the unit tests I've got for the flag helper functions.
